lwt/cohttp_lwt_unix_test.ml

```diff
@@ -0,0 +1,68 @@
+open Lwt
+open OUnit
+open Cohttp
+open Cohttp_lwt_unix
+
+type spec = Request.t -> Cohttp_lwt_body.t
+  -> (Response.t * Cohttp_lwt_body.t) Lwt.t
+
+type async_test = unit -> unit Lwt.t
+
+let port = ref 9193
+
+let next_port () =
+  let current_port = !port in
+  incr port;
+  current_port
+
+let response_sequence responses =
+  let xs = ref responses in
+  fun _ _ ->
+    match !xs with
+    | x::xs' ->
+      xs := xs';
+      x
+    | [] -> Lwt.fail_with "response_sequence: Server exhausted responses"
+
+let temp_server ?port spec callback =
+  let port = match port with
+    | None -> next_port ()
+    | Some p -> p in
+  let server = Server.make ~callback:(fun _ req body -> spec req body) () in
+  let uri = Uri.of_string ("http://0.0.0.0:" ^ (string_of_int port)) in
+  let server_ready = Lwt_unix.sleep 0.3 in (* XXX hackish *)
```

Comment:
Are you finding that there's a race between `Server.create` and connections being accepted?  It ought to block until the socket is open and listening.